### PR TITLE
[MRG] Linear Neural Networks/Concise Implementation of Linear Regression

### PR DESCRIPTION
**Pull request by xiaotinghe**

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-zh/PR-554/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-554/

**Comment by mli** [comment]
Job d2l-zh/PR-554/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-554/

**Comment by mli** [comment]
Job d2l-zh/PR-554/3 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-554/

File: chapter_linear-networks/linear-regression-concise.md [hunk 1]
```diff
@@ -1,13 +1,14 @@
 # 线性回归的简洁实现
 :label:`sec_linear_concise`
 
-过去几年对深度学习的广泛兴趣激发了公司、学术界和业余爱好者开发了各种成熟的开源框架，以实现基于梯度的学习算法的重复工作的自动化。在 :numref:`sec_linear_scratch` 中，我们仅依赖 (i) 数据存储和线性代数的张量；(ii) 计算梯度的自动分化。在实践中，由于数据迭代器、损失函数、优化器和神经网络层非常常见，现代库也为我们实现了这些组件。
+在过去的几年里，出于对深度学习强烈的兴趣，各个公司、学者和业余爱好者开发了各种成熟的开源框架。通过这些框架可以自动化实现基于梯度的学习算法中重复性的工作。
```
Comment (astonzhang) [inline chapter_linear-networks/linear-regression-concise.md:4]: （i）这里的括号要用全角，其他PR所有地方都要改

File: chapter_linear-networks/linear-regression-concise.md [hunk 2]
```diff
@@ -82,26 +84,26 @@ next(iter(data_iter))
 
 ## 定义模型
 
-当我们在 :numref:`sec_linear_scratch` 中实现从头开始的线性回归时，我们明确地定义了模型参数，并对计算进行编码，以使用基本的线性代数运算生成输出。你应该知道如何做到这一点。但是，一旦你的模型变得更加复杂，一旦你几乎每天都需要这样做，你会很高兴得到帮助。这种情况类似于从头开始编写自己的博客。这样做一次或两次是有益的和有启发性的，但如果每次你需要一个博客，你花了一个月的时间来重新发明轮子，你将是一个糟糕的 Web 开发人员。
+当我们在 :numref:`sec_linear_scratch` 中实现线性回归时，我们明确定义了模型参数变量，并编写了计算过程的代码，这样通过基本的线性代数运算得到输出。但是，如果模型变得更加复杂，而且当您几乎每天都需要实现模型时，你会很乐于简化这个过程。这种情况类似于从头开始编写自己的博客。做一两次是有益的、有启发性的，但如果每次你每需要一个博客就花一个月的时间重新发明轮子，那你将是一个糟糕的web开发者。
```
Comment (astonzhang) [inline chapter_linear-networks/linear-regression-concise.md:87]: 您 -》 你

Comment (astonzhang) [reply chapter_linear-networks/linear-regression-concise.md:87]: web -> Web

File: chapter_linear-networks/linear-regression-concise.md [hunk 3]
```diff
@@ -82,26 +84,26 @@ next(iter(data_iter))
...
 
-对于标准操作，我们可以使用框架的预定义图层，这使我们能够特别关注用于构建模型的图层，而不必专注于实现。我们将首先定义一个模型变量 `net`，它将引用 `Sequential` 类的一个实例。`Sequential` 类为将链接在一起的多个图层定义了一个容器。给定输入数据时，`Sequential` 实例将其传递到第一层，然后将输出作为第二层的输入传递等。在下面的样本中，我们的模型只包含一个层，因此我们不需要 `Sequential`。但是，由于我们几乎所有未来的模型都会涉及多个层次，我们将使用它只是为了让您熟悉最标准的工作流程。
+对于标准操作，我们可以使用框架的预定义好的层。这使我们只需关注使用哪些层来构造模型，而不必关注实现细节。我们首先定义一个模型变量`net`，它是一个 `Sequential` 类的实例。 `Sequential` 类为串联在一起的多个层定义了一个容器。当给定输入数据， `Sequential` 实例将其传入到第一层，然后将第一层的输出作为第二层的输入，依此类推。在下面的例子中，我们的模型只包含一个层，因此实际上不需要`Sequential`。但是由于以后几乎所有的模型都将涉及到多层，我们使用`Sequential`会让你熟悉最标准的工作流。
 
-回想一下单层网络的体系结构，如 :numref:`fig_single_neuron` 所示。该图层被认为是 * 完全连接 *，因为它的每个输入通过矩阵矢量乘法连接到每个输出。
+回顾 :numref:`fig_single_neuron` 中的单层网络架构，这一单层被称为 *全连接层*，因为它的每一个输入都通过矩阵乘法连接到它的每个输出。
```
Comment (astonzhang) [inline chapter_linear-networks/linear-regression-concise.md:91]: 术语后面要有英文注释

File: chapter_linear-networks/linear-regression-concise.md [hunk 4]
```diff
@@ -127,18 +129,20 @@ net.add(tf.keras.layers.Dense(1))
...
 ## 初始化模型参数
 
-在使用 `net` 之前，我们需要初始化模型参数，例如线性回归模型中的权重和偏差。深度学习框架通常有一种预定义的方式来初始化参数。在这里，我们指定每个权重参数应从平均值 0 和标准差 0.01 的正态分布随机采样。偏差参数将初始化为零。
+在使用`net`之前，我们需要初始化模型参数，如线性回归模型中的权重和偏差。
+深度学习框架通常有一种预定义的方法来初始化参数。
+在这里，我们指定每个权重参数应该从均值为0，标准差为0.01的正态分布中随机采样，偏差参数将初始化为零。
 
 :begin_tab:`mxnet`
-我们将从 MxNet 导入 `initializer` 模块。本模块为模型参数初始化提供了各种方法。粘合剂使 `init` 可作为访问 `initializer` 软件包的快捷方式（缩写）。我们只指定如何通过调用 `init.Normal(sigma=0.01)` 来初始化权重。默认情况下，偏置参数初始化为零。
+我们从 MxNet 导入 `initializer` 模块。这个模块提供了各种模型参数初始化方法。Gluon将 `init` 作为访问 `initializer` 包的快捷方式。我们只通过调用 `init.Normal(sigma=0.01)` 来指定如何初始化权重。默认情况下，偏差参数初始化为零。
```
Comment (astonzhang) [inline chapter_linear-networks/linear-regression-concise.md:137]: MXNet

File: chapter_linear-networks/linear-regression-concise.md [hunk 5]
```diff
@@ -160,29 +164,33 @@ net.add(tf.keras.layers.Dense(1, kernel_initializer=initializer))
 ```
 
 :begin_tab:`mxnet`
-上面的代码可能看起来简单，但你应该注意到这里发生了一些奇怪的事情。我们正在初始化网络的参数，即使 Gluon 还不知道输入将具有多少维度！它可能是 2，如我们的例子，也可能是 2000 年。Gluon 让我们摆脱这一点，因为在幕后，初始化实际上是 * 推迟 *。只有当我们第一次尝试通过网络传递数据时，才会进行真正的初始化。只要小心记住，由于参数尚未被初始化，我们无法访问或操作它们。
+上面的代码可能看起来很简单，但是您应该注意到这里的一个细节：我们正在为网络初始化参数，而Gluon还不知道输入将有多少维!
```
Comment (astonzhang) [inline chapter_linear-networks/linear-regression-concise.md:167]: 您-》你，PR其他地方都要改

File: chapter_linear-networks/linear-regression-concise.md [hunk 6]
```diff
@@ -228,17 +236,20 @@ trainer = torch.optim.SGD(net.parameters(), lr=0.03)
 trainer = tf.keras.optimizers.SGD(learning_rate=0.03)
 ```
 
-## 培训
+## 训练
 
-您可能已经注意到，通过深度学习框架的高级 API 表达我们的模型需要相对较少的代码行。我们不必单独分配参数，定义我们的损失函数，或实现小批次随机梯度下降。一旦我们开始使用更复杂的模型，高级 API 的优势将大幅增长。但是，一旦我们有所有的基本部分，训练循环本身就与我们从头开始实施所有内容时所做的非常相似。
+通过深度学习框架的高级api来实现我们的模型只需要相对较少的代码行。
```
Comment (astonzhang) [inline chapter_linear-networks/linear-regression-concise.md:241]: api要大写

File: chapter_linear-networks/linear-regression-concise.md [hunk 7]
```diff
@@ -228,17 +236,20 @@ trainer = torch.optim.SGD(net.parameters(), lr=0.03)
...
+我们不必单独分配参数、不必定义我们的损失函数，也不必手动实现小批量随机梯度下降。
+当我们需要更复杂的模型时，高级api的优势将大大增加。
```
Comment (astonzhang) [inline chapter_linear-networks/linear-regression-concise.md:243]: api要大写

File: chapter_linear-networks/linear-regression-concise.md [hunk 8]
```diff
@@ -228,17 +236,20 @@ trainer = torch.optim.SGD(net.parameters(), lr=0.03)
...
+当我们有了所有的基本组件，训练循环与我们从零开始实现所有东西时所做的非常相似。
```
Comment (astonzhang) [inline chapter_linear-networks/linear-regression-concise.md:244]: “训练循环”有没有更好的翻译？

File: chapter_linear-networks/linear-regression-concise.md [hunk 9]
```diff
@@ -228,17 +236,20 @@ trainer = torch.optim.SGD(net.parameters(), lr=0.03)
...
 
-刷新内存：对于一些时代，我们将完全传递数据集（`train_data`），迭代地抓取一个小批输入和相应的地面真实标签。对于每个小批次，我们将通过以下仪式：
+回顾一下：在迭代周期里，我们将完整遍历一次数据集（`train_data`），迭代地获取一小批量的输入和相应的标签。对于每一个小批量，我们会进行以下程序:
```
Comment (astonzhang) [inline chapter_linear-networks/linear-regression-concise.md:246]: “迭代地获取”？

File: chapter_linear-networks/linear-regression-concise.md [hunk 10]
```diff
@@ -303,48 +316,48 @@ b = net.get_weights()[1]
...
+## 总结
 
 :begin_tab:`mxnet`
-* 使用 Gluon，我们可以更简洁地实现模型。
-* 在 Gluon 中，`data` 模块提供了数据处理工具，`nn` 模块定义了大量的神经网络层，`loss` 模块定义了许多常见的损耗函数。
-* MxNet 的模块 `initializer` 为模型参数初始化提供了各种方法。
-* 维度和存储是自动推断的，但请注意不要在初始化参数之前尝试访问参数。
+* 我们可以使用Gluon更简洁地实现模型。
+* 在 Gluon 中`data` 模块提供了数据处理工具，`nn` 模块定义了大量的神经网络层，`loss` 模块定义了许多常见的损失函数。
+* MxNet 的 `initializer` 模块提供了各种模型参数初始化方法。
```
Comment (astonzhang) [inline chapter_linear-networks/linear-regression-concise.md:324]: MXNet

File: chapter_linear-networks/linear-regression-concise.md [hunk 11]
```diff
@@ -303,48 +316,48 @@ b = net.get_weights()[1]
...
+* 维度和存储可以自动推断，但请注意不要在初始化参数之前尝试访问参数。
```
Comment (astonzhang) [inline chapter_linear-networks/linear-regression-concise.md:325]: 去掉“请”

File: chapter_linear-networks/linear-regression-concise.md [hunk 12]
```diff
@@ -303,48 +316,48 @@ b = net.get_weights()[1]
...
 :end_tab:
 
 ## 练习
 
 :begin_tab:`mxnet`
-1. 如果我们用 `l = loss(output, y).mean()` 替换 `l = loss(output, y)`，我们需要将 `trainer.step(batch_size)` 更改为 `trainer.step(1)`，以使代码的行为相同。为什么？
-1. 查看 MxNet 文档，了解模块 `gluon.loss` 和 `init` 中提供了哪些丢失函数和初始化方法。以 Huber 的损失补偿损失。
+1. 如果我们用 `l = loss(output, y).mean()` 替换 `l = loss(output, y)`。我们为了使代码的行为相同，需要将 `trainer.step(batch_size)` 更改为 `trainer.step(1)`，这是为什么？
+1. 查看 MxNet 文档，了解模块 `gluon.loss` 和 `init` 中提供了哪些损失函数和初始化方法。用Huber损失来代替。
```
Comment (astonzhang) [inline chapter_linear-networks/linear-regression-concise.md:345]: MXNet

**Comment by mli** [comment]
Job d2l-zh/PR-554/4 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-554/

**Comment by mli** [comment]
Job d2l-zh/PR-554/5 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-554/

File: chapter_linear-networks/linear-regression-concise.md (hunk 8 above)
Comment (astonzhang) [reply chapter_linear-networks/linear-regression-concise.md:244]: training loop -> 训练部分代码

**Comment by mli** [comment]
Job d2l-zh/PR-554/6 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-554/

**Comment by mli** [comment]
Job d2l-zh/PR-554/7 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-554/

**Pull request merged**